### DISCUSSION TITLE
Where can I find logs for Navidrome in the Docker version? It's for the purposes of forming a good bug report.

### DISCUSSION THREAD

**Discussion by IO101010** [navidrome/navidrome#3294] [Q&A]
I'm running the latest Docker image and sometimes when a song ends, the next song never loads. I'm left with silence.

I'd like to find a way to reproduce this consistently to make a good bug report. Besides checking the web inspector for JS errors, how do I check logs in the case of Docker?

**Comment by deluan** [comment]
`docker logs` or `docker compose logs`. Check the help for options.

> sometimes when a song ends, the next song never loads. I'm left with silence.

Which client? If it is Navidrome's UI, also look for errors in the browser console.

**Reply by IO101010** [reply]
Thanks. Safari web browser so yes, web UI. I'll check for errors there.

**Comment by SickProdigy** [comment]
You might want to enable LogLevel = 'DEBUG' to show more details of logs.

**Comment by IO101010** [comment]
**Steps to reproduce**
I'm having trouble reproducing this issue on a device I can debug. Here are the steps to reproduce it on mobile, in case it gives anyone ideas:

Requirements: iPhone 12 with iOS 17 or 18.
1: Open navidrome from the Safari mobile browser on iOS.
2: Play an album or playlist and switch away from Safari (either switch to a different app or lock the screen).

Expected result: When the current song finishes playing, the next one begins.

Actual result: Playback stops at the end of the current song. The name of the next song does appear on the lockscreen and time 0:00.

When I bring Safari to the foreground music resumes playing immediately.

**Logging**
ND_LOGLEVEL=info
The log below is the output for the problem situation. "The Adventure Continues" played fine as it was the first song. I put Safari to the background once it started playing (no lockscreen, just homescreen). Audio playback then stopped. You can see that the next song on the album "Scribing Beneath First Light" is logged, however it did not actually begin to play on the device.

```
time="2024-10-16T11:10:50Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=1448 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082096 title="The Adventure Continues" transcoding=false user=i
time="2024-10-16T11:10:50Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=1448 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082097 title="The Adventure Continues" transcoding=false user=i
time="2024-10-16T11:10:51Z" level=info msg="Now Playing" artist="Michael Land" player="NavidromeUI [Safari/iOS]" requestId=b939f8d5d71a/5ZSPU6SaP3-082098 title="The Adventure Continues" user=i
time="2024-10-16T11:11:10Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=579 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082099 title="Scribing Beneath First Light" transcoding=false user=i
time="2024-10-16T11:11:10Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=579 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082100 title="Scribing Beneath First Light" transcoding=false user=i
time="2024-10-16T11:11:10Z" level=info msg=Scrobbled artist="Michael Land" requestId=b939f8d5d71a/5ZSPU6SaP3-082101 timestamp="2024-10-16 11:10:51.13 +0000 UTC" title="The Adventure Continues" user=i
time="2024-10-16T11:11:29Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=579 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082104 title="Scribing Beneath First Light" transcoding=false user=i
time="2024-10-16T11:11:29Z" level=info msg="Streaming file" artist="Michael Land" bitRate=0 cached=false format=raw originalBitRate=579 originalFormat=flac requestId=b939f8d5d71a/5ZSPU6SaP3-082105 title="Scribing Beneath First Light" transcoding=false user=i
```

**Attempts to reproduce elsewhere:**
I've tried to reproduce this on Safari on desktop in the hope that I could debug the lient side. By making the window small enough, the responsive design changes to what I usually see on the phone (the cool turntable design). But the problem does not occur.

I've also tried using the iOS simulator on the desktop where I can potentially use tools to get client-side debug info, but the problem does not occur.

Next: I'll borrow an android device from someone soon and see if I can reproduce the issue there.

**Reply by metalheim** [reply]
Afaik, this is a known Safari-mobile issue for a long time that Apple is not willing to fix (it might be related to Apple disliking PWAs bc they want to protect their appStore revenue cut and they have their own music streaming service)

**Comment by IO101010** [comment]
That makes sense. Thanks, I can move on from disappointment to acceptance :) I just love the rotating cover art so much and no client apps have that on iOS. On 16 Oct 2024, at 22:23, Stephan Wahlen ***@***.***> wrote:﻿
Afaik, this is a known Safari-mobile issue for a long time that Apple is not willing to fix (it might be related to Apple disliking PWAs bc they want to protect their appStore revenue cut and they have their own music streaming service)
